File: server-rails-single/config/initializers/rhino.rb
# frozen_string_literal: true

# Rhino Configuration — SINGLE-TENANT variant.
#
# No Organization / Role / UserRole. Domain models are owned by user_id
# (Project directly; Task/Comment inherit ownership via their parent Project).
# Label is a shared global catalog (no owner). Ownership isolation is enforced
# by the Scopes::{Model}Scope default scopes, not by any organization.
Rhino.configure do |config|
  # ---------------------------------------------------------------
  # Models (no organizations / roles)
  # ---------------------------------------------------------------
  config.model :projects, "Project"
  config.model :tasks, "Task"
  config.model :comments, "Comment"
  config.model :labels, "Label"

  # ---------------------------------------------------------------
  # Route Groups
  # ---------------------------------------------------------------
  # One `default` route group: empty prefix, no domain, no org middleware. Auth
  # is the standard global /api/auth/* set (the legacy auth routes map to the
  # :default group). All models are served by it.
  config.route_group :default,
    prefix: "",
    models: :all

  # ---------------------------------------------------------------
  # Auth / Group membership
  # ---------------------------------------------------------------
  # No groups/roles to enforce — membership enforcement stays off. Permissions
  # are not role-based: the User model omits HasPermissions, so the policy falls
  # through to "allow" and ownership is enforced entirely by the model scopes.
  config.auth = { enforce_group_membership: false }

  # ---------------------------------------------------------------
  # Nested Operations
  # ---------------------------------------------------------------
  config.nested = {
    path: "nested",
    max_operations: 50,
    allowed_models: nil
  }

  # ---------------------------------------------------------------
  # Test Framework
  # ---------------------------------------------------------------
  config.test_framework = "rspec"
end

File: server-laravel-single/config/rhino.php
<?php

return [
    // Single-tenant: NO organizations / roles / user_roles models at all.
    'models' => [
        'comments' => \App\Models\Comment::class,
        'labels' => \App\Models\Label::class,
        'projects' => \App\Models\Project::class,
        'tasks' => \App\Models\Task::class,
    ],

    // One route group: 'default'. Empty prefix, no domain, no org middleware.
    // Auth is the standard global /api/auth/* set. Ownership is enforced by the
    // per-model query scopes in app/Models/Scopes (user_id == auth id), so there
    // is no organization resolution here.
    'route_groups' => [
        'default' => [
            'prefix' => '',
            'middleware' => [],
            'models' => '*',
        ],
    ],

    // Membership enforcement is off (no groups/roles in this variant).
    'auth' => [
        'enforce_group_membership' => false,
    ],

    'invitations' => [
        'expires_days' => env('INVITATION_EXPIRES_DAYS', 7),
        'allowed_roles' => null,
    ],
    'nested' => [
        'path' => 'nested',
        'max_operations' => 50,
        'allowed_models' => null,
    ],
    'client_path' => env('RHINO_CLIENT_PATH'),
    'mobile_path' => env('RHINO_MOBILE_PATH'),
    'test_framework' => 'pest',
];
